libafl/src/executors/afl_args.rs

```diff
@@ -0,0 +1,220 @@
+//! Share implementation of afl style arguments
+
+use alloc::{borrow::ToOwned, vec::Vec};
+use std::{
+    ffi::{OsStr, OsString},
+    path::Path,
+};
+
+use libafl_bolts::fs::{InputFile, get_unique_std_input_file};
+
+/// How to deliver input to an external program
+/// `StdIn`: The target reads from stdin
+/// `File`: The target reads from the specified [`InputFile`]
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub enum InputLocation {
+    /// Mutate a commandline argument to deliver an input
+    Arg {
+        /// The offset of the argument to mutate
+        argnum: usize,
+    },
+    /// Deliver input via `StdIn`
+    #[default]
+    StdIn,
+    /// Deliver the input via the specified [`InputFile`]
+    /// You can use specify [`InputFile::create(INPUTFILE_STD)`] to use a default filename.
+    File {
+        /// The file to write input to. The target should read input from this location.
+        out_file: InputFile,
+    },
+}
+
+/// The main implementation trait of afl style arguments handling
+pub trait HasAflStyleTargetArguments: Sized {
```

Comment:
I don't think this should be a `Has` trait, since it's not a single thing but ... just a trait

Reply:
AflTargetArgs ?

Reply:
Make sense

Reply:
Done, waiting for CI